lib/deploy_pin/deployment_state.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+module DeployPin
+  # The mixin that extends DeployPin module with
+  # @ongoing_deployment? & @pending_deployment? methods
+  module DeploymentState
+    ONGOING = 'ongoing'
+    PENDING = 'pending'
+    CACHE_KEY = 'deploy_pin:deployment'
+
+    DEPLOYMENT_OVER_TASK = %(
+      # no_file_task
+      # -1:%<group>s:recurring
+      # task_title: Cleanup DeployPin.state
+
+      DeployPin.send(:deployment_over!)
+      ).strip
+
+    DEPLOYMENT_ONGOING_TASK = %(
+      # no_file_task
+      # 0:%<group>s:recurring
+      # task_title: Set DeployPin.state to 'ongoing'
+
+      DeployPin.send(:deployment_ongoing!)
+      ).strip
+
+    DEPLOYMENT_PENDING_TASK = %(
+      # no_file_task
+      # 0:%<group>s:recurring
+      # task_title: Set DeployPin.state to 'pending'
+
+      DeployPin.send(:deployment_pending!)
+      ).strip
+
+    def ongoing_deployment?
```

Comment:
is it possible to make deployment_state an enumerator?

Reply:
What is the purpose for that?